lib/Dialect/FIRRTL/Transforms/InferWidths.cpp

```diff
@@ -1596,12 +1596,12 @@ LogicalResult InferenceMapping::mapOperation(Operation *op) {
       })
       .Case<MuxPrimOp, Mux2CellIntrinsicOp>([&](auto op) {
         auto *sel = getExpr(op.getSel());
-        constrainTypes(sel, solver.known(1));
+        constrainTypes(solver.known(1), sel, /*imposeUpperBounds=*/true);
```

Comment:
The constraint change here goes from adding `sel >= 1` to `sel <= 1` which isn't used for solving but is checked for correctness.